### ISSUE TITLE
Serious difficulty building on Windows

### ISSUE THREAD

**Issue by mfeemster** [steve-o/openpgm#43]
In order to build with MSVC, one must install CMake, Perl, Python and a merge patching tool. This takes many hours to accomplish something rather trivial: making a VS solution.

Is there a way legitimate VS support can be added with some basic solution and project files that build out of the box without forcing the builder to jump through so many hoops?

**Comment by steve-o** [comment]
Yes, the awful C98 patch system should be cleaned up with only support for MSVC 2013 and 2015.

CMake support was specifically introduced for all the different MSVC versions and can create project, solution, or NMake files as required.  As Windows XP is EOL some toolkit support can probably be simplified as that was quite troublesome.

The Python and Perl pieces are more challenging, the generated code can be checked into the repository leaving the problem to generating the version header.  Some more investigation will be required.

**Comment by steve-o** [comment]
Working on MSVC 2013 as CMake stable does not support 2015 U1.

**Comment by steve-o** [comment]
Resolved with ea25a4c.  Example build:

```
mkdir build
cd build
"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" x64
"C:\Program Files (x86)\CMake\bin\cmake.exe" -G "Visual Studio 12 2013 Win64" ..
-- The C compiler identification is MSVC 18.0.40629.0
...
msbuild /nologo /property:Configuration=Debug ALL_BUILD.vcxproj
...
```
